base/show.jl

```diff
@@ -593,14 +593,14 @@ function show(io::IO, p::Pair)
     iocompact = IOContext(io, :compact => get(io, :compact, true))
     isdelimited(io, p) && return show_default(iocompact, p)
 
-    typeinfos = gettypeinfos(io, p)
-    isdelimited(iocompact, p.first) || print(io, "(")
-    show(IOContext(iocompact, :typeinfo => typeinfos[1]), p.first)
-    isdelimited(iocompact, p.first) || print(io, ")")
+    io1, io2 = (IOContext(iocompact, :typeinfo => ti) for ti in gettypeinfos(io, p))
```

Comment:
Could you rewrite this without the generator? It seems silly to use it for exactly 2 items.

Reply:
is the `for` loop better?